### PR TITLE
Add host target ports to host fact in purefa_facts

### PR DESCRIPTION
**Pull request by sdodsley**
##### SUMMARY
Add the collection of host target ports connected to a host object

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
purefa_facts

### CODE REVIEW

File: lib/ansible/modules/storage/purestorage/purefa_facts.py [hunk 1]
```diff
@@ -590,7 +590,9 @@ def generate_host_dict(array):
             'iqn': hosts[host]['iqn'],
             'wwn': hosts[host]['wwn'],
             'personality': array.get_host(hostname,
-                                          personality=True)['personality']
+                                          personality=True)['personality'],
+            'target_port': array.get_host(hostname,
+                                          all=True)[0]['target_port']
```
Comment (sile16) [inline lib/ansible/modules/storage/purestorage/purefa_facts.py]: The 0 is assuming there is at least 1 results.  I get an Error List index out of range when I run this.

Here are the raw results returned from my array.  As you see some have empty target ports which generates this error.  And some have multiple ports.  I think the return type should just be a list and pass that directly back as is.  So, just remove the [0] from the line:

```
[
  {
    "host_wwn": null,
    "name": "esxi1",
    "host_iqn": "iqn.1998-01.com.vmware:esxi1-32e248c8",
    "vol": "FlashStack-Datastore2",
    "hgroup": "FlashStack-Cluster",
    "target_port": [],
    "lun": 254
  },
  {
    "host_wwn": "20000025B5000A02",
    "name": "esxi1",
    "host_iqn": null,
    "vol": "FlashStack-Datastore2",
    "hgroup": "FlashStack-Cluster",
    "target_port": [
      "CT1.FC2",
      "CT0.FC2"
    ],
    "lun": 254
  },
```

**Review by sile16 — CHANGES_REQUESTED** [review]
Getting error index out of range.  see the line review.

File: lib/ansible/modules/storage/purestorage/purefa_facts.py (hunk 1 above)
Comment (sdodsley) [reply lib/ansible/modules/storage/purestorage/purefa_facts.py]: Removing the 0 doesn't work. We have to pass an index. Working on a fix.

File: lib/ansible/modules/storage/purestorage/purefa_facts.py [hunk 2]
```diff
@@ -585,12 +585,17 @@ def generate_host_dict(array):
     hosts = array.list_hosts()
     for host in range(0, len(hosts)):
         hostname = hosts[host]['name']
+        tports = []
+        host_all_info = array.get_host(hostname, all=True)
+        if host_all_info:
+            tports = host_all_info[0]['target_ports']
```
Comment (sile16) [inline lib/ansible/modules/storage/purestorage/purefa_facts.py]: Now getting a "KeyError" on "target_ports" .  should be "target_port" . ? 

line 591, in generate_host_dict\r\nKeyError: 'target_ports'\r\n"

```
{
    "host_wwn": "20000025B5000A02",
    "name": "esxi1",
    "host_iqn": null,
    "vol": "FlashStack-Datastore2",
    "hgroup": "FlashStack-Cluster",
    "target_port": [
      "CT1.FC2",
      "CT0.FC2"
    ],
    "lun": 254
```

**Review by sile16 — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/storage/purestorage/purefa_facts.py (hunk 2 above)
Comment (sdodsley) [reply lib/ansible/modules/storage/purestorage/purefa_facts.py]: Now that is what you get for rushing :)

**Review by sile16 — APPROVED** [review]
works now for me.

**Comment by sile16** [comment]
shipit

**Pull request merged**